webapp/channels/src/packages/mattermost-redux/src/reducers/entities/preferences.ts

```diff
@@ -62,8 +80,37 @@ function myPreferences(state: Record<string, PreferenceType> = {}, action: AnyAc
     }
 }
 
+function userPreferences(state: Record<string, PreferencesType> = {}, action: AnyAction) {
+    switch (action.type) {
+    case PreferenceTypes.RECEIVED_USER_ALL_PREFERENCES:
+        return setAllUserPreferences(action.data);
+
+    case PreferenceTypes.RECEIVED_USER_PREFERENCES: {
+        const nextState = {...state};
+
+        const data = action.data as PreferenceType[];
+        if (action.data && data.length > 0) {
+            const userID = data[0].user_id;
+            nextState[userID] = nextState[userID] ? {...nextState[userID]} : {};
+
+            for (const preference of action.data) {
+                nextState[preference.user_id][getKey(preference)] = preference;
+            }
+        }
+
+        return nextState;
+    }
+
+    case UserTypes.LOGOUT_SUCCESS:
+        return {};
+    default:
+        return state;
+    }
+}
+
 export default combineReducers({
 
     // object where the key is the category-name and has the corresponding value
     myPreferences,
+    userPreferences,
```

Comment:
I'm storing it in redux store so that I can reuse the selectors used in all the individual setting components. Otherwise its gonna involve a lot of refactoring.